### ISSUE TITLE
SPProductUpdate: wrong detect version / patch not installed

### ISSUE THREAD

**Issue by paololec** [PowerShell/SharePointDsc#994]
#### Details of the scenario you tried and the problem that is occurring
Try to use SPProductUpdate with "sts2019-kb4461513-fullfile-x64-glb.exe" , two server SP, one CU apply on first server already, when try SPProductUpdate on second server it NOT EXECUTED because wrong evalution of right version using "Get-SPDscFarmVersionInfo"

#### Verbose logs showing the problem
Update has version 16.0.10338.12107
The lowest version of any SharePoint component is 16.0.10338.12107 <- incorrect !!!

#### Suggested solution to the issue
Improve "Get-SPDscFarmVersionInfo" 
$patchableUnitInfo report info from multipleServers so use $patchableUnitInfo.LatestPatch.Version.ToString() to detect "currentversion" is incorrect.

First simple resolution (I think..) using info from specific server to detect right version:
$patchableUnitInfo.LatestPatchOnServer($server.Id).Version.ToString()

Other method:
Maybe using $patchableUnitInfo.Patches (or PatchesByVersion) there is "ServersMissingThis" that detect the missing CU...

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
```powershell
 SPProductUpdate InstallCU_STS
        {
            SetupFile            = $ConfigurationData.SharePoint.Settings.CUPath + "\sts2019-kb4461513-fullfile-x64-glb.exe"
            ShutdownServices     = $true
            Ensure               = "Present"
            PsDscRunAsCredential = $Credssp_admin
        }
```

#### The operating system the target node is running
Microsoft Windows Server 2019 - en-US
 
#### Version of SharePoint that is used (e.g. SharePoint 2016)
SharePoint 2019
#### Version and build of PowerShell the target node is running
5.1.17763.134      
#### Version of the DSC module that was used ('dev' if using current dev branch)
3.0

**Comment by ykuijs** [comment]
@paololec Thanks for reporting. We will have to investigate this issue.

FYI: We are working on a revised method for detecting patch levels, so we might have to skip fixing this issue because the new method will be released.

**Comment by andikrueger** [comment]
This issue should be resolved by the new method. The updated SPProductUpdate resource and additional functions to check the install status of an update can be found here: https://github.com/PowerShell/SharePointDsc/pull/938

You can find more information about the chances made in this PR in #679 .

**Comment by ykuijs** [comment]
Will be fixed by PR #1046

**Comment by ykuijs** [comment]
Closed via PR #1046
